R/tm_g_scatterplotmatrix.R

```diff
@@ -267,217 +283,222 @@ ui_g_scatterplotmatrix <- function(id, ...) {
+        varnames <- varname_w_label(cols_names, ANL, wrap_width = 20)
 
-      # check character columns. If any, then those are converted to factors
-      check_char <- vapply(ANL[, cols_names], is.character, logical(1))
-      if (any(check_char)) {
-        qenv <- teal.code::eval_code(
-          qenv,
-          substitute(
-            expr = ANL <- ANL[, cols_names] %>%
-              dplyr::mutate_if(is.character, as.factor) %>%
-              droplevels(),
-            env = list(cols_names = cols_names)
+        # check character columns. If any, then those are converted to factors
+        check_char <- vapply(ANL[, cols_names], is.character, logical(1))
+        if (any(check_char)) {
+          qenv <- teal.code::eval_code(
+            qenv,
+            substitute(
+              expr = ANL <- ANL[, cols_names] %>%
+                dplyr::mutate_if(is.character, as.factor) %>%
+                droplevels(),
+              env = list(cols_names = cols_names)
+            )
           )
-        )
-      } else {
-        qenv <- teal.code::eval_code(
-          qenv,
-          substitute(
-            expr = ANL <- ANL[, cols_names] %>%
-              droplevels(),
-            env = list(cols_names = cols_names)
+        } else {
+          qenv <- teal.code::eval_code(
+            qenv,
+            substitute(
+              expr = ANL <- ANL[, cols_names] %>%
+                droplevels(),
+              env = list(cols_names = cols_names)
+            )
           )
-        )
-      }
+        }
 
 
-      # create plot
-      if (add_cor) {
-        shinyjs::show("cor_method")
-        shinyjs::show("cor_use")
-        shinyjs::show("cor_na_omit")
+        # create plot
+        if (add_cor) {
+          shinyjs::show("cor_method")
+          shinyjs::show("cor_use")
+          shinyjs::show("cor_na_omit")
 
-        qenv <- teal.code::eval_code(
-          qenv,
-          substitute(
-            expr = {
-              g <- lattice::splom(
-                ANL,
-                varnames = varnames_value,
-                panel = function(x, y, ...) {
-                  lattice::panel.splom(x = x, y = y, ...)
-                  cpl <- lattice::current.panel.limits()
-                  lattice::panel.text(
-                    mean(cpl$xlim),
-                    mean(cpl$ylim),
-                    get_scatterplotmatrix_stats(
-                      x,
-                      y,
-                      .f = stats::cor.test,
-                      .f_args = list(method = cor_method, na.action = cor_na_action)
-                    ),
-                    alpha = 0.6,
-                    fontsize = 18,
-                    fontface = "bold"
-                  )
-                },
-                pch = 16,
-                alpha = alpha_value,
-                cex = cex_value
+          qenv <- teal.code::eval_code(
+            qenv,
+            substitute(
+              expr = {
+                g <- lattice::splom(
```

Comment:
plot binding in `qenv` is `g` while `renderPlot` uses `"plot"`. `print()` of the graph is executed twice.
```suggestion
                g <- lattice::splom(
```

Reply:
Corrected here: e1cce8d0